tests/integration_tests/functional/test_feat_parity.py

```diff
@@ -0,0 +1,395 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Tests for the verifying features exposed by CPUID and MSRs by various CPU templates."""
+
+import pytest
+
+from conftest import _test_images_s3_bucket
+from framework.artifacts import ArtifactCollection, ArtifactSet
+from framework.matrix import TestMatrix, TestContext
+from framework.builder import MicrovmBuilder
+import framework.utils_cpuid as cpuid_utils
+import host_tools.network as net_tools
+
+
+# All existing CPU templates available on Intel
+INTEL_TEMPLATES = ["C3", "T2", "T2CL", "T2S"]
+# All existing CPU templates available on AMD
+AMD_TEMPLATES = ["T2A"]
+# All existing CPU templates
+ALL_TEMPLATES = INTEL_TEMPLATES + AMD_TEMPLATES
+# CPU templates designed to provide instruction set feature parity
+INST_SET_TEMPLATES = ["T2A", "T2CL"]
+
+
+def get_supported_templates():
+    """
+    Returns the list of CPU templates supported by the platform.
+    """
+    vendor = cpuid_utils.get_cpu_vendor()
+    if vendor == cpuid_utils.CpuVendor.INTEL:
+        return INTEL_TEMPLATES
+    if vendor == cpuid_utils.CpuVendor.AMD:
+        return AMD_TEMPLATES
+    return []
```

Comment:
Do we need to check if x86_64 here?

Reply:
I was wondering `cpuid_utils.CpuVendor` is valid on aarch64 for checking this. If it is I have no other concerns.

Reply:
I can see it used in other tests. I don't expect for it to be broken.